code/modules/reagents/chemistry_machinery/chem_master.dm

```diff
@@ -141,7 +141,23 @@
 			if(length(label) < 3)
 				loaded_pill_bottle.maptext_label = label
 				loaded_pill_bottle.update_icon()
+	else if(href_list["setcolor"])
+		// Checking for state changes
+		if(!loaded_pill_bottle)
+			return
+
+		if(!Adjacent(usr))
+			return
 
+		var/num = copytext(reject_bad_text(input(usr,"Color? (1-12)", "Set \the [src]'s color", "")), 1, 3)
```

Comment:
A color is fine to internally be represented as a number, but to a player a number doesn't equate to a color.

- Make this a `tgui_input_list` instead where you list out different color choices.

- Also since this code thats now getting re-used in multiple places, consider making it a proc for a pill bottle instead so you can re-use the code.

- Also you don't need to use `\the` or `\The` they'll automatically get filled in. It doesn't really matter, but Harry has been trying to just remove these all generally.